polyfill/test/zoneddatetime.mjs

```diff
@@ -730,6 +730,23 @@ describe('ZonedDateTime', () => {
     });
   });
 
+  describe('Maths', () => {
+    const earlier = ZonedDateTime.from('1976-11-18T15:23:30.123456789-03:00[America/Santiago]');
+    const later = ZonedDateTime.from('2019-10-29T10:46:38.271986102-03:00[America/Santiago]');
+    ['years', 'months', 'weeks', 'days', 'hours', 'minutes', 'seconds'].forEach((largestUnit) => {
```

Comment:
I'd think you need 2-4 more variations of this test to test reversibility with DST involved.  If there's disambiguation at the point where you're done adding days and need to add the time, then behavior will vary based on the direction of the operation.  It's possible that those tests already exist, so can ignore this comment if so. But there's a lot of confusing logic around the interplay between dates, times, and DST and it's easy to get it wrong.  I think that you'll know you have good values of `earlier` and `later` when `a.add(d).equals(b)===true` but `b.subtract(d).equals(a)===false` and the results are 1 or 2 hours apart.

For variations, it probably needs one for the transition on the `earlier` side, and one on the `later` side, and then a test of both forward and backward transitions.

Reply:
I don't think I'll get this done tonight, but let's open an issue tomorrow to add more test coverage.

Reply:
→ #1140